test/functional/azure_rm_test.rb

```diff
@@ -4,19 +4,18 @@ module ForemanAzureRM
   class ComputeResourcesControllerTest < ActionController::TestCase
     tests ::ComputeResourcesController
 
-    setup { Fog.mock! }
-    teardown { Fog.unmock! }
-
     test "should create a compute resource and return edit page" do
-      test_client = Fog::Resources::AzureRM.new(
+      test_sdk = ForemanAzureRM::AzureSDKAdapter.new(
         tenant_id:       '',
         client_id:       '',
         client_secret:   '',
-        subscription_id: '',
-        :environment     => 'AzureCloud')
+        subscription_id: ''
+      )
 
-      ForemanAzureRM::AzureRM.any_instance.stubs(:rg_client).returns(test_client)
-      test_client.stubs(:list_resource_groups).returns([])
+      ForemanAzureRM::AzureRM.any_instance.stubs(:sdk).returns(test_sdk)
+      test_resource_client = Object.new
+      test_sdk.stubs(:resource_client).returns(test_resource_client)
+      test_resource_client.stubs(:rgs).returns([])
```

Comment:
Is it a common case that resource groups would return an empty array? If it is, I would suggest adding another test with a non-empty array and validates that we are displaying the groups correctly. If this is an edge case - we can probably modify the current test to work with non-empty array.